UnityProject/Assets/Scripts/UI/Nuke/GUI_Nuke.cs

```diff
@@ -0,0 +1,37 @@
+﻿using UI;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class GUI_Nuke : MonoBehaviour
+{
+    //setting up the window
+    public GameObject buttonPrefab;
+    private static readonly Color banColor = new Color(255f / 255f, 103f / 255f, 103f / 255f);
+    private static readonly Color infoColor = new Color(200f / 255f, 200f / 255f, 200f / 255f);
+    public Text title;
+    public InputField code;
+    //what needs to happen when the button is clicked
+    public void BtnOk()
+    {
+        SoundManager.Play("Click01");
+        UIManager.Display.nukeWindow.SetActive(false);
+        GameObject nuke = GameObject.Find("nuke");
+        NukeInteract ni = (NukeInteract)nuke.GetComponent("NukeInteract");
+        string codestring = code.text;
+        //send the code to NukeInteract
+        ni.validate(codestring);
```

Comment:
A NetworkMessage should be created (Derive from ClientMessage, see other messages for examples) and then used here to take the inputted code and send it to the server to be validated. This will stop people from hacking as in this PR the code is also synced on clients.